seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/dag/physical/SubPlan.java

```diff
@@ -231,6 +231,9 @@ private PipelineStatus getPipelineEndState() {
                 errorByPhysicalVertex.compareAndSet(
                         null, checkpointCoordinatorState.getThrowableMsg());
             }
+            if (getPipelineState().equals(PipelineStatus.FAILING)) {
+                pipelineStatus = PipelineStatus.FAILED;
+            }
```

Comment:
Please add some comment for this special logic.

Reply:
done